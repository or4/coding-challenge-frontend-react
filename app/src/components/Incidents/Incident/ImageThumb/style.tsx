import styled, { StyledComponent } from 'styled-components';
import { media } from 'ui/utils';

import prevewSvg from './assets/preview.svg';
import { IProps } from '.';

interface IContainerProps {
    isPreview: boolean;
}

export const Container: StyledComponent<'div', any, IContainerProps, never> = styled.div`
    ${media.mobile`
        display: ${(props: IContainerProps) => (props.isPreview ? 'none' : 'inherit')};
		padding-bottom: 10px;
  	`}

    ${media.tabletNDesktop`
		width: 100px;
		height: 100px;
		min-width: 100px;
		min-height: 100px;

		display: flex;
		align-items: center;
		justify-content: center;
		box-sizing: border-box;
		border-radius: 4px;
		border: 1px solid rgba(0, 0, 0, 0.05);
    	background: rgba(0, 0, 0, 0.02);
	`}
`;

export const Image = styled.div`
    background-image: url(${(props: IProps) => props.imageUrl});
    background-repeat: no-repeat;
    background-size: cover;

    ${media.mobile`
		padding-top: 100%;
		width: 100%;
  	`}

    ${media.tabletNDesktop`
		width: 90px;
		height: 90px;
	`}
`;

export const Preview = styled.div`
    background-image: url(${prevewSvg});
    background-repeat: no-repeat;
    background-size: cover;

    width: 55px;
    height: 32px;

    ${media.mobile`
		display: none;
  	`}
`;
