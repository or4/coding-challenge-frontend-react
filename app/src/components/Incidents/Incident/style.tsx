import styled from 'styled-components';
import { media } from 'ui/utils';

export const Container = styled.div`
    display: flex;
    cursor: pointer;
    padding-top: 10px;
    padding-bottom: 10px;
    font-size: 14px;
    outline: none;

    ${media.mobile`
		flex-direction: column;
		border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  	`}

    ${media.tabletNDesktop`
    	min-height: 100px;
	`}

    &:hover {
        background: rgba(0, 0, 0, 0.03);
    }
`;

export const TextContainer = styled.div`
    display: flex;
    flex-direction: column;

    ${media.tabletNDesktop`
        padding: 0 15px;
    `}
`;

export const IncidentTitle = styled.div`
    font-weight: 600;

    ${media.mobile`
        padding-bottom: 5px;
    `}

    ${media.tabletNDesktop`
        padding-bottom: 10px;
    `}
`;

export const IncidentDescription = styled.div`
    ${media.mobile`
        padding-bottom: 5px;
    `}

    ${media.tabletNDesktop`
        padding-bottom: 10px;
    `}
`;

export const IncidentDetails = styled.div``;
