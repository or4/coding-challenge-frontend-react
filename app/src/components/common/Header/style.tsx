import styled from 'styled-components';

import { media } from 'ui/utils';

export const Container = styled.div`
    display: flex;
    align-items: center;
    cursor: pointer;

    ${media.mobile`
        height: 60px;
        margin-bottom: 15px;
    `}

    ${media.tablet`
        height: 100px;
    `}

    ${media.desktop`
        height: 140px;
    `}
`;

export const Logo = styled.img`
    width: 64px;
    height: 64px;

    ${media.tabletNDesktop`
        padding: 0 18px;
    `}
`;

export const TextWrapper = styled.div`
    padding-left: 15px;
`;

export const Title = styled.div`
    font-weight: 600;

	${media.mobile`
    	font-size: 18px;
  	`}

	${media.tablet`
    	font-size: 32px;
	`}

    ${media.desktop`
    	font-size: 48px;
	`}
`;

export const Description = styled.div`
	font-weight: 600;

	${media.mobile`
    	font-size: 14px;
  	`}

	${media.tablet`
    	font-size: 16px;
	`}

    ${media.desktop`
    	font-size: 32px;
	`}
`;
