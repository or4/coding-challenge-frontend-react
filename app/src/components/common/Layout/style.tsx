import styled from 'styled-components';

import { media } from 'ui/utils';

export const Container = styled.div`
    margin: 0 auto 15px;

    ${media.mobile`
		padding: 15px;
  	`}

	${media.tablet`
		padding: 15px;
	`}

    ${media.desktop`
		max-width: 940px;
	`}
`;
