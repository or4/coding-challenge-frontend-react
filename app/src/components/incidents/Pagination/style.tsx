import styled from 'styled-components';

import { PageButton } from 'ui/PageButton';

export const Container = styled.div`
    text-align: center;
    margin: 15px 0;
`;

export const Button = styled(PageButton)`
    margin-right: calc(15px + 3px);

    &:last-child {
        margin-right: 0;
    }
`;
