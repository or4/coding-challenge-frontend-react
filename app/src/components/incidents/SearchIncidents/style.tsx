import styled from 'styled-components';
import { media } from 'ui/utils';

export const QueryInput = styled.input`
    min-width: 240px;
    padding: 4px 8px;
    flex-grow: 1;
    margin: 5px 5px;

    ${media.mobileNTablet`
		width: 100%;
	`}
`;
