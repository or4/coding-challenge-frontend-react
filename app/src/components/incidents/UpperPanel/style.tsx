import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    margin: -5px -5px;
`;
