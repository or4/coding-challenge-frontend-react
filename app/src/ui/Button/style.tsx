import styled from 'styled-components';

import { IProps as IButtonProps } from '.';

export const Container = styled.div`
    display: inline-block;
    box-sizing: border-box;
    color: rgba(0, 0, 0, 0.9);
    background-color: #ffffff;
    cursor: ${(props: IButtonProps) => (props.canClick ? 'pointer' : 'default')};
    box-shadow: 3px 3px 0px rgba(0, 0, 0, 0.9);
    border: 3px solid rgba(0, 0, 0, 0.9);

    line-height: 14px;
    height: 30px;
    user-select: none;

    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    max-width: 100%;

    font-size: 14px;
    padding: 5px 10px;

    &:hover {
        background-color: rgba(0, 0, 0, 0.05);
    }

    ${(props: IButtonProps) =>
        props.canClick
            ? `
                &:active {
                    background-color: rgba(0, 0, 0, 0.9);
                    box-shadow: 0px 0px 0px rgba(0, 0, 0, 0.9);
                    border: 3px solid transparent;
                    color: #ffffff;
                }`
            : ''}
`;
