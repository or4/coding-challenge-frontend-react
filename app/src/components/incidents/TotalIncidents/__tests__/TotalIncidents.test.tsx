import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { TotalIncidents } from '..';
import { Container } from '../style';

describe('TotalIncidents', () => {
    it('should correct render', () => {
        const tree = renderer.create(<TotalIncidents value={4} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should contain container', () => {
        const wrapper = shallow(<TotalIncidents value={1} />);

        expect(wrapper.find(Container)).toHaveLength(1);
    });

    it('should pass total pages prop', () => {
        const totalPages = 4;
        const wrapper = shallow(<TotalIncidents value={totalPages} />);

        expect(wrapper.text()).toEqual(`Total: ${totalPages}`);
    });

    it('should render when total pages is not defined', () => {
        const wrapper = shallow(<TotalIncidents />);

        expect(wrapper.text()).toEqual(`Total: 0`);
    });

    it('should render when total pages equal zero', () => {
        const wrapper = shallow(<TotalIncidents value={0} />);

        expect(wrapper.text()).toEqual(`Total: 0`);
    });
});
