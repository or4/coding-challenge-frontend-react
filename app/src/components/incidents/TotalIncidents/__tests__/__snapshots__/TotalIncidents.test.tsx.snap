// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalIncidents should correct render 1`] = `
.c0 {
  min-height: 40px;
  line-height: 40px;
  text-align: right;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
  className="c0"
  data-test-id="total-incidents"
>
  Total: 4
</div>
`;
